easybuild/tools/filetools.py

```diff
@@ -408,7 +416,18 @@ def extract_file(fn, dest, cmd=None, extra_options=None, overwrite=False, forced
 
     run.run_cmd(cmd, simple=True, force_in_dry_run=forced)
 
-    return find_base_dir()
+    # note: find_base_dir also changes into the base dir!
+    base_dir = find_base_dir()
+
+    # if changing into obtained directory is not desired,
+    # change back to where we came from (unless that was a non-existing directory)
+    if not change_into_dir:
+        if cwd is None:
+            _log.warning("Can't change back to non-existing directory after extracting %s in %s", fn, dest)
```

Comment:
Are you sure this shouldn't be an fatal error?

Reply:
Yeah, it doesn't make much sense if this isn't fatal... fixed in 4bb149d31